pkg/kubernetes/nulecule.js

```diff
@@ -0,0 +1,257 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "kubernetes/config"
+], function($, cockpit, config) {
+    "use strict";
+
+    var nulecule = { };
+    var _ = cockpit.gettext;
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "nulecule")
+            console.debug.apply(console, arguments);
+    }
+
+    function failure(ex) {
+        console.warn(ex);
+    }
+
+    /**
+     * NuleculeClient
+     *
+     */
+    function NuleculeClient() {
+        var self = this;
+        var http = cockpit.http(5000);
+        var version = '0.1.1';
+        var status = "";
+        var answers = {};
+        var install_dir = "";
+
+        self.create_tmp = function create_tmp() {
+            var process = cockpit.spawn(['/bin/sh', '-s']).input("mktemp -p /tmp -d APP_ENTITY.XXXX");
+            return process;
+        };
+
+        self.writeAnswerfile = function writeAnswerfile(install_dir, data) {
+            return cockpit.file(install_dir + "/answers.conf").replace(JSON.stringify(data));
+        };
+
+        self.get_version = function get_version() {
+            return version;
+        };
+
+        self.get_answers = function get_answers() {
+            return answers;
+        };
+
+        self.set_answers = function set_answers(ans) {
+            answers = ans;
+        };
+
+        self.get_install_dir = function get_install_dir() {
+            return install_dir;
+        };
+
+        self.loadAnswersfile = function loadAnswersfile(statusl) {
+            statusl.forEach(function(item) {
+                if (item.status_message.indexOf("answers.conf") > -1)
+                    answers = item.status_data;
+                    debug("answer file contents : " + String(answers));
+            });
+            return answers;
+        };
+
+        self.get_statuslist = function get_statuslist() {
+            var dfd = $.Deferred();
+            var response;
+            var req = http.get("/atomicapp-run/api/v" + version + "/status")
+                    .done(function(data) {
+                        req = null;
+                        try {
+                            response = JSON.parse(data);
+                            dfd.resolve(response.items);
+                        } catch(ex) {
+                            failure(ex);
+                            dfd.reject(ex);
+                        }
+                    })
+                    .fail(function(ex) {
+                        failure(ex);
+                        req = null;
+                        dfd.reject(ex);
+                    });
+            var promise = dfd.promise();
+            promise.cancel = function cancel() {
+                req.cancel();
```

Comment:
This will try to dereference a null pointer if the http request has already completed. Since you clear `req` above.